### DISCUSSION TITLE
Appending B.1: pipx not found after installation

### DISCUSSION THREAD

**Discussion by hoffimar** [daneah/publishing-python-packages#44] [Q&A]
After installing asdf and python-launcher on macos (see appendix A), I installed pipx as in appendix B (`py -m pip install pipx`). After starting a new shell, pipx is not found in $PATH.
I see it was installed in my .venv for my repository, should it be installed somewhere else while I'm not in a virtual environment? Is the installation instruction in B.1 correct or am I missing some precondition?

**Comment by daneah** [comment]
> I see it was installed in my .venv for my repository, should it be installed somewhere else while I'm not in a virtual environment? Is the installation instruction in B.1 correct or am I missing some precondition?

Sorry you ran into trouble! You have the right hunch for a fix. It sounds like you created (or had) a project before following the appendix instructions and ran the installation there. The intent is to install pipx into the globally-enabled Python installation so that it is available on your path. You could do this by running the same installation command for pipx from e.g. your home directory.

**Reply by hoffimar** [reply]
Hmm not yet working. I installed it into asdf's python dir:
```
> py -3.11 -m pip install pipx  
Collecting pipx
  Using cached pipx-1.2.0-py3-none-any.whl (57 kB)
Requirement already satisfied: argcomplete>=1.9.4 in /Users/<...>/.asdf/installs/python/3.11.3/lib/python3.11/site-packages (from pipx) (3.0.8)
Requirement already satisfied: packaging>=20.0 in /Users/<...>/.asdf/installs/python/3.11.3/lib/python3.11/site-packages (from pipx) (23.1)
Requirement already satisfied: userpath>=1.6.0 in /Users/<...>/.asdf/installs/python/3.11.3/lib/python3.11/site-packages (from pipx) (1.8.0)
Requirement already satisfied: click in /Users/<...>/.asdf/installs/python/3.11.3/lib/python3.11/site-packages (from userpath>=1.6.0->pipx) (8.1.3)
Installing collected packages: pipx
Successfully installed pipx-1.2.0
```
I am using zsh, installed asdf and python-launcher. What do I need to make pipx available on the PATH?
So far I can of course use the tools from the venv (without using pipx), but I was wondering how to do it without messing around with the PATH manually.
I assume that `/Users/<...>/.asdf/installs/python/3.11.3/bin/` should have been put into my PATH automatically.

**Reply by daneah** [reply]
> I assume that `/Users/<...>/.asdf/installs/python/3.11.3/bin/` should have been put into my PATH automatically.

That is indeed my expectation; perhaps an issue up at the asdf level?
